### PR TITLE
Add expired nodes to zombie list and send peer update on deletion

### PR DESCRIPTION
**Pull request by alphadose**

### CODE REVIEW

**Review by abhishek9686 — COMMENTED** [review]
can't node expiration be handled during check-in?

**Comment by alphadose** [comment]
the `CheckZombies` function appends the node id to a slice which gets cleaned up all together at once by the `ManageZombies` goroutine

**Review by abhishek9686 — APPROVED** [review]
can't node expiration be handled during check-in?

File: logic/zombie.go [hunk 1]
```diff
@@ -80,11 +81,13 @@ func ManageZombies(ctx context.Context) {
...
 						continue
 					}
-					if time.Since(node.LastCheckIn) > time.Minute*ZOMBIE_DELETE_TIME {
+					if time.Since(node.LastCheckIn) > time.Minute*ZOMBIE_DELETE_TIME || time.Now().After(node.ExpirationDateTime) {
 						if err := DeleteNode(&node, true); err != nil {
 							logger.Log(1, "error deleting zombie node", zombies[i].String(), err.Error())
 							continue
 						}
+						node.Action = models.NODE_DELETE
+						peerUpdate <- &node
```
Comment (mattkasun) [inline logic/zombie.go:90]: why use a channel here; why not just call mq.NodeUpdate ?

File: main.go [hunk 2]
```diff
@@ -191,7 +191,15 @@ func runMessageQueue(wg *sync.WaitGroup) {
 	mq.SetupMQTT()
 	ctx, cancel := context.WithCancel(context.Background())
 	go mq.Keepalive(ctx)
-	go logic.ManageZombies(ctx)
+	go func() {
```
Comment (mattkasun) [inline main.go:194]: why do we need this go routine.

**Review by mattkasun — COMMENTED** [review]
questions

**Comment by theguy951357** [comment]
will wait on changes

File: logic/zombie.go (hunk 1 above)
Comment (abhishek9686) [reply logic/zombie.go:90]: i guess it's because of cyclic import calling `mq` from logic pkg

Comment (alphadose) [reply logic/zombie.go:90]: @abhishek9686 yes

File: main.go (hunk 2 above)
Comment (alphadose) [reply main.go:194]: for sending peer update on node deletion by `ManageZombies`. main package for no cyclic dependency

**Comment by theguy951357** [comment]
I set the node to the year 2000 and watch the logs. I have not seen the expired node go into the zombie list.

**Review by theguy951357 — APPROVED** [review]
expired node went away
pass

**Pull request merged**